scheduler/util.go

```diff
@@ -184,6 +195,20 @@ func diffSystemAllocsForNode(
 				node.Status == structs.NodeStatusDisconnected &&
 				exist.ClientStatus == structs.AllocClientStatusRunning {
 
+				// Handle the case where the client updates its allocs before
+				// sending a heartbeat. The alloc will be set to running, but
+				// the client is still considered to be disconnected.
```

Comment:
Same question as above: I thought this was prevented in the node endpoint changes?

Since the eval's WaitIndex is _at least_ its CreatedIndex, and it will have been created by the node heartbeat at the earliest, I don't think there's any chance of a "the statestore might not have applied the raft log the scheduler needs to make the right decision yet" issue, so as long as the node endpoint prevents this logic I'm not sure why we also need to prevent it here?